lms/services/lti_grading/_v11.py

```diff
@@ -49,22 +54,52 @@ def read_result(self, grading_id) -> GradingResult:
         return result
 
     def record_result(self, grading_id, score=None, pre_record_hook=None, comment=None):  # noqa: ARG002
-        request = {"resultRecord": {"sourcedGUID": {"sourcedId": grading_id}}}
-
-        if score is not None:
-            request["resultRecord"]["result"] = {
-                "resultScore": {"language": "en", "textString": score}
-            }
-
+        request = self._record_score_payload(score=score, user_grading_id=grading_id)
         if pre_record_hook:
             request = pre_record_hook(score=score, request_body=request)
 
-        self._send_request({"replaceResultRequest": request})
+        self._send_request(
+            self.application_instance,
+            self.line_item_url,
+            {"replaceResultRequest": request},
```

Comment:
We can not include `replaceResultRequest` in `_record_score_payload` because the `pre_record_hook` doesn't expect it. I don't want to refactor that now because the ideal refactor is to remove the hook enterly.